www/front_src/src/Resources/Graph/Performance/AnomalyDetection/AnomalyDetectionEnvelopeThreshold.tsx

```diff
@@ -0,0 +1,213 @@
+import { equals, prop, isNil } from 'ramda';
+import { Threshold } from '@visx/threshold';
+import { curveBasis } from '@visx/curve';
+import { ScaleLinear, ScaleTime } from 'd3-scale';
+import { LinePath } from '@visx/shape';
+
+import { useTheme } from '@mui/material/styles';
+
+import { TimeValue, Line } from '../models';
+import { getTime, getYScale } from '../timeSeries';
+
+import AnomalyDetectionShapeCircle from './AnomalyDetectionShapeCircle';
+import { CustomFactorsData } from './models';
+
+interface Props {
+  data?: CustomFactorsData;
+  graphHeight: number;
+  leftScale: ScaleLinear<number, number>;
+  regularLines: Array<Line>;
+  rightScale: ScaleLinear<number, number>;
+  secondUnit: string;
+  thirdUnit: string;
+  timeSeries: Array<TimeValue>;
+  xScale: ScaleTime<number, number>;
+}
+
+interface ParamsDiff {
+  factors: CustomFactorsData;
+  item: TimeValue;
+  metricLower: string;
+  metricUpper: string;
+}
+
+const AnomalyDetectionEnvelopeThreshold = ({
+  secondUnit,
+  regularLines,
+  xScale,
+  leftScale,
+  rightScale,
+  thirdUnit,
+  timeSeries,
+  graphHeight,
+  data,
+}: Props): JSX.Element => {
+  const theme = useTheme();
+
+  const [
+    {
+      metric: metricY1,
+      unit: unitY1,
+      invert: invertY1,
+      lineColor: lineColorY1,
+    },
+  ] = regularLines.filter((item) => equals(item.name, 'Upper Threshold'));
+
+  const [
+    {
+      metric: metricY0,
+      unit: unitY0,
+      invert: invertY0,
+      lineColor: lineColorY0,
+    },
+  ] = regularLines.filter((item) => equals(item.name, 'Lower Threshold'));
+
+  const y1Scale = getYScale({
+    hasMoreThanTwoUnits: !isNil(thirdUnit),
+    invert: invertY1,
+    leftScale,
+    rightScale,
+    secondUnit,
+    unit: unitY1,
+  });
+
+  const y0Scale = getYScale({
+    hasMoreThanTwoUnits: !isNil(thirdUnit),
+    invert: invertY0,
+    leftScale,
+    rightScale,
+    secondUnit,
+    unit: unitY0,
+  });
+
+  const getXPoint = (timeValue): number => xScale(getTime(timeValue)) as number;
+  const getY1Point = (timeValue): number =>
+    y1Scale(prop(metricY1, timeValue)) ?? null;
+  const getY0Point = (timeValue): number =>
+    y0Scale(prop(metricY0, timeValue)) ?? null;
+
+  if (data) {
```

Comment:
I would extract in a specific component as it contains a lot of code

Reply:
btw, what is `data`?

Reply:
data contains the slider data that was put by user , (current value , default value..) to calculate the shape of the graph depending on this data